inc/save.php

```diff
@@ -96,8 +116,12 @@ function save() {
  * }
  */
 function validate_line( $line, $line_number ) {
```

Comment:
```suggestion
function validate_line( $line, $line_number, $has_placeholder_record = false ) {
```

Will need a docblock changes but I can't add it as it's outside the diff.